containers/hbadeals/rsem2hbadeals.R

```diff
@@ -152,25 +158,48 @@ all      <- colnames(countsData)
 toRemove <- c("transcript_id","gene_id")
 toKeep   <- all [!all  %in% toRemove]
 
-# Apply exclusion criterion 1: Filter out transcripts with cumsum across samples lower than the N, cohort size
-# message("\nApplying exclusion criterion 1:  ( discard transcripts with very low count across samples) ..")
-countsData <- countsData[rowSums(countsData[,toKeep] > 0 ) >= sample_size, ]
-# message("Done!")
+# Apply exclusion criterion 1:
+# Discard transcripts with zero counts across samples (zeroes tolerance threshold on minority class is set to zeroes_threshold)
+message("\nApplying exclusion criterion 1: discard transcripts with zero counts across samples (zeroes tolerance threshold on minority class is set to ", zeroes_threshold, ")")
+
+N_controls       <- length(metaData[[status_colname]][metaData[[status_colname]] == 1])
+N_cases          <- length(metaData[[status_colname]][metaData[[status_colname]] == 2])
+N_minority_class <- min(N_controls, N_cases)
+
+message('N controls       : ', N_controls       , appendLF = FALSE)
+message('N cases          : ', N_cases          , appendLF = FALSE)
+message('N minority class : ', N_minority_class , appendLF = FALSE)
+
+control_ids <- metaData[[sample_colname]][metaData[[status_colname]] == 1]
+cases_ids   <- metaData[[sample_colname]][metaData[[status_colname]] == 2]
+
+essage('dim(countsData)[1] before transcript exclusion: ', dim(countsData)[1])
+
+message("\nApplying exclusion criterion 1: discard transcripts with zero counts across samples (zeroes tolerance threshold on minority class is set to ", zeroes_tolerance_threshold, ")")
+countsData <- countsData[rowSums(countsData[, colnames(countsData) == control_ids ] > 0) >= N_minority_class * zeroes_tolerance_threshold, ]
+message('dim(countsData)[1] after applying transcript exclusion based on control: ', dim(countsData)[1])
+
+countsData <- countsData[rowSums(countsData[, colnames(countsData) == cases_ids   ] > 0) >= N_minority_class * zeroes_tolerance_threshold, ]
+message('dim(countsData)[1] after applying transcript exclusion based on cases: ', dim(countsData)[1])
+
+
+message("Done!")
```

Comment:
Implements updated filtering as discussed.

Reply:
⚠️ CAUGHT ANOTHER MISTAKE:

Typo `essage` -> `message`

Reply:
> ⚠️ CAUGHT ANOTHER MISTAKE:
> 
> Typo `essage` -> `message`

Fixed in https://github.com/TheJacksonLaboratory/coronavirus-nf/pull/3/commits/3f4139b129ea42c6dfa6bcbca3f1924bcdbb51bf .